middleware/realip.go

```diff
@@ -9,9 +9,11 @@ import (
 	"strings"
 )
 
-var trueClientIP = http.CanonicalHeaderKey("True-Client-IP")
-var xForwardedFor = http.CanonicalHeaderKey("X-Forwarded-For")
-var xRealIP = http.CanonicalHeaderKey("X-Real-IP")
+var DefaultRealIPHeaders = []string{
```

Comment:
I don't love making package-level variables as any dependency could change the values and create side effects. I'd rather make it a const, or unexported and folks can pass their own with the new method

Reply:
you cannot declare an array/slice as const, so this would be close as possible.

```golang
func DefaultRealIPHeaders() []string {
	return []string{
		"True-Client-IP", // Cloudflare Enterprise plan
		"X-Real-IP",
		"X-Forwarded-For",
	}
}
```

or

```golang
func RealIPFromHeaders(realIPHeaders []string, addDefaultHeaders bool) func(http.Handler) http.Handler {
...
```

but is it before or after the custom headers. i would always export default header.

Reply:
If I choose to have a full control of the headers, I wouldn't mind passing them all explicitly.

This is what I'd expect to do:

```diff
- r.Use(middleware.RealIP)
+ r.Use(middleware.RealIPFromHeaders("CF-Connecting-IP", "X-Real-IP", "X-Forwarded-For"))
```

I think we can keep the var unexported.

(even tho I don't mind the function returning it)